### PR TITLE
Allow output for user requested data in parallel runs.

### PR DESCRIPTION
**Pull request by blattms**
This PR further aligns parallel output with the one done in a serial. Previously outputting data in addition to the cell data registered with in `SimulationDataContainer` was not possible in a parallel run. Now we use the `data::Solution` map when gathering data. As this also gets the user requested cell data, we output this in parallel now, too.

Norne seems to run fine. Will test with Model 2 also.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins run norne please

**Comment by blattms** [comment]
Model 2 ran smoothly with this in parallel

File: opm/autodiff/ParallelDebugOutput.hpp [hunk 1]
```diff
@@ -367,10 +403,10 @@ namespace Opm
                 }
 
                 // write all cell data registered in local state
-		for (const auto& pair : localState_.cellData()) {
```
Comment (blattms) [inline opm/autodiff/ParallelDebugOutput.hpp]: Yes that is intended as all the array are in cellData anyway. After gathering the data we extract that data from `globalCellData_` to `globalState_` using `solutionToSim`.

Also note that the globalState_ is only needed to support rudimentary matlab ouput in a parallel run. This output is (and was always) broken as the velocities will be wrong.

Comment (blattms) [inline opm/autodiff/ParallelDebugOutput.hpp]: The update of globalReservoirState_ happens in [line 612 of ParallelDebugOutput.hpp](/blattms/opm-simulators/blob/00440ab3449baf915732f8761ef2fa7d48a3486d/opm/autodiff/ParallelDebugOutput.hpp#L612)

Comment (atgeirr) [inline opm/autodiff/ParallelDebugOutput.hpp]: I did not notice the solutionToSim() call, but now I get it, looks correct.

**Comment by atgeirr** [comment]
Thanks for the patch, merging!

**Pull request merged**